m2e-maven-runtime/org.eclipse.m2e.maven.runtime/src/main/java/org/eclipse/m2e/internal/maven/listener/M2EMavenBuildDataBridge.java

```diff
@@ -180,7 +182,7 @@ public static MavenBuildConnection prepareConnection(String label, Consumer<Mave
 			try (ServerSocketChannel s = server; SocketChannel readChannel = server.accept()) {
 				ByteBuffer buffer = ByteBuffer.allocate(512);
 				StringBuilder message = new StringBuilder();
-				while (readChannel.read(buffer.clear()) >= 0) {
+				while (readChannel.read(buffer) >= 0) {
```

Comment:
The buffer had to be cleared at the end of the loop, not only ones before (the initial clear is actually not needed, but was there because it fit in the control flow).